xlators/protocol/server/src/server.c

```diff
@@ -862,6 +862,16 @@ server_reconfigure(xlator_t *this, dict_t *options)
     GF_FREE(this->ctx->statedump_path);
     this->ctx->statedump_path = gf_strdup(statedump_path);
 
+    statedump_path = NULL;
+    GF_OPTION_RECONF("volspec-directory", statedump_path, options, path,
+                     do_auth);
+    if (!statedump_path) {
+        goto do_auth;
+    }
+    gf_path_strip_trailing_slashes(statedump_path);
+    GF_FREE(conf->volfile_dir);
+    conf->volfile_dir = gf_strdup(statedump_path);
+
```

Comment:
I think it's cleaner this way:

```suggestion
    if (statedump_path) {
        gf_path_strip_trailing_slashes(statedump_path);
        GF_FREE(conf->volfile_dir);
        conf->volfile_dir = gf_strdup(statedump_path);
    }
```

Also, we should check errors from `gf_strdup()`.

Reply:
ack. done (the first part)